aten/src/ATen/templates/Operators.h

```diff
@@ -2,6 +2,13 @@
 
 // ${generated_comment}
 
+#ifdef TORCH_ASSERT_NO_OPERATORS
+#error This change adds a dependency on native_functions.yaml,             \
+  meaning the file will need to be re-compiled every time an operator      \
+  is changed or added. Consider if your change would be better placed in   \
+  another file, or if a more specific header might achieve the same goal.
+#endif
```

Comment:
So, @gchanan mentioned that if the problem is just the methods on Tensor, as long as you just don't edit methods in native_functions.yaml (hard, I know) this shouldn't induce recompilation. So then *technically* you don't need to split Tensor and TensorBase; you just need to make sure you aren't accidentally including Functions.h. I appreciate that the situation on the ground, however, might make it a bit easier, but I'm wondering if there is something else flawed about this analysis.

Reply:
At the moment, `Tensor.h` does depend on all signatures through `Operators.h` but in theory of course that could be fixed. However, that's still more dependency than is necessary for most operator kernels so either way I think this is valuable.